added panel.hpp

josephalvis opened the pull request:
simple change for overlapping windows

ChrisThrasher commented:
Can you add this header to cxxcurses.hpp?

josephalvis commented:
Hi Chris,

The reason I didn't is because it uses the additional libpanel (which for me came with ncurses).

I don't think it's a problem to simply include the header without using panels but I didn't want to break anything with unresolved symbols.

If the panel class is used, libpanel has to be linked. Is this a problem?

ChrisThrasher commented:
> The reason I didn't is because it uses the additional libpanel
 
Oh okay I wasn't aware.
 
 > If the panel class is used, libpanel has to be linked. Is this a problem?
 
I think there ought to be a second CMake target that depends on the current target but also links to libpanel so that no users has to manually link to libpanel themselves.

josephalvis commented:
Hi Chris,

I added a new target cxxcurses_panel/cxxcurses::panel. I'm not sure about the name but it seems to be working. 

Unfortunately FindCurses.cmake does not look for libpanel so I added a find_library call as well.

I also added sample/panel.cpp with a really dumb(but working) example. It's basically just a tweaked main.cpp.

ChrisThrasher commented:
I'm having issues building your branch:

```
Consolidate compiler generated dependencies of target cxxcurses_panel_sample
[ 25%] Building CXX object sample/CMakeFiles/cxxcurses_panel_sample.dir/panel.cpp.o
In file included from /Users/Thrasher/Projects/cxxcurses/sample/panel.cpp:1:
In file included from /Users/Thrasher/Projects/cxxcurses/include/cxxcurses/cxxcurses.hpp:18:
/Users/Thrasher/Projects/cxxcurses/include/cxxcurses/panel.hpp:53:86: error: use of undeclared identifier 'ceiling_panel'
    [[nodiscard]] static auto ceiling(raw::screen_ptr sp) -> raw::panel_ptr { return ceiling_panel(sp); }
                                                                                     ^
/Users/Thrasher/Projects/cxxcurses/include/cxxcurses/panel.hpp:54:86: error: use of undeclared identifier 'ground_panel'
    [[nodiscard]] static auto ground (raw::screen_ptr sp) -> raw::panel_ptr { return ground_panel(sp); }
                                                                                     ^
2 errors generated.
make[2]: *** [sample/CMakeFiles/cxxcurses_panel_sample.dir/panel.cpp.o] Error 1
make[1]: *** [sample/CMakeFiles/cxxcurses_panel_sample.dir/all] Error 2
make: *** [all] Error 2
```

ChrisThrasher commented:
```cmake
if(CURSES_NEED_WIDE)
    find_library(CURSES_PANEL_LIB "panelw")
else()
    find_library(CURSES_PANEL_LIB "panel")
endif()
```

I think this may be removed since we're not setting `CURSES_NEED_WIDE` ourselves.
https://cmake.org/cmake/help/latest/module/FindCurses.html
 > New in version 3.10: Set CURSES_NEED_WIDE to TRUE before the find_package(Curses) call if unicode functionality is required.
